addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -22,24 +22,21 @@ def _gc_unpin_livechat_sessions(self):
         sessions_to_be_unpinned.write({'unpin_dt': fields.Datetime.now()})
         self.env['bus.bus']._sendmany([(member.partner_id, 'discuss.channel/unpin', {'id': member.channel_id.id}) for member in sessions_to_be_unpinned])
 
-    def _partner_data_to_store(self, store: Store, fields=None):
+    def _to_store(self, store: Store, **kwargs):
+        super()._to_store(store, **kwargs)
+        for member in self.filtered(lambda m: m.channel_id.channel_type == "livechat"):
+            # sudo: mail.channel - reading livechat channel to check whether current member is a bot is allowed
+            store.add(
+                "discuss.channel.member",
+                {
+                    "id": member.id,
+                    "is_bot": member.partner_id
```

Comment:
Note: moving `is_bot` to member and not partner, as it is compute from/linked to the channel of the member.